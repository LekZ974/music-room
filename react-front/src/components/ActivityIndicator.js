import React from 'react';
import { ActivityIndicator as ActivityIndicatorComponent, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';

const styles = StyleSheet.create({
  loading: {
    position: 'absolute',
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  bgColor: {
    backgroundColor: '#ffffff',
  },
});

const ActivityIndicator = ({ animating, transparent }) => {
  return (
    <ActivityIndicatorComponent
      style={[styles.loading, !transparent && styles.bgColor]}
      animating={animating}
      size="large"
    />
  );
};

ActivityIndicator.defaultProps = {
  transparent: false,
};

ActivityIndicator.propTypes = {
  animating: PropTypes.bool.isRequired,
  transparent: PropTypes.bool,
};

export default ActivityIndicator;
