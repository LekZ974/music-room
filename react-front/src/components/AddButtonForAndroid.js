import React, { Component } from 'react';
import { withNavigation } from 'react-navigation';

import { TouchableOpacity, Animated, View, StyleSheet } from 'react-native';
import FAwesomeIcon from 'react-native-vector-icons/FontAwesome';

import PropTypes from 'prop-types';
import { Theme } from '../native-base-theme/default_theme';

/** constantes * */

const bigBubbleSize = 100;

/** Style * */
const style = StyleSheet.create({
  bigBubble: {
    justifyContent: 'center',
    alignItems: 'center',
    height: bigBubbleSize,
    width: bigBubbleSize,
    borderRadius: bigBubbleSize / 2,
    top: -30,
  },
});

/** class * */
class AddButtonForIos extends Component {
  constructor(props) {
    super(props);
    this.animatedValue = new Animated.Value(0);
    this.topLeftValue = new Animated.Value(0);
    this.topCenterValue = new Animated.Value(0);
    this.topRightValue = new Animated.Value(0);
  }

  render() {
    const { navigation } = this.props;

    const goToCreateRoom = () => {
      navigation.navigate('CreateRoomScreen');
    };

    const springValue = Animated.add(
      Animated.add(this.topLeftValue, this.topRightValue),
      this.topCenterValue,
    );

    return (
      <View>
        <Animated.View
          style={[
            style.bigBubble,
            {
              backgroundColor: Theme.palette.primary,
              transform: [
                {
                  rotateZ: springValue.interpolate({
                    inputRange: [0, 1, 2, 3],
                    outputRange: ['-45deg', '-45deg', '0deg', '45deg'],
                  }),
                },
                {
                  scaleY: springValue.interpolate({
                    inputRange: [0, 0.65, 1, 1.65, 2, 2.65, 3],
                    outputRange: [1, 1.1, 1, 1.1, 1, 1.1, 1],
                  }),
                },
              ],
            },
          ]}
        >
          <TouchableOpacity
            hitSlop={{
              left: 20,
              right: 20,
              top: 20,
              bottom: 20,
            }}
            onPress={goToCreateRoom}
          >
            <Animated.View
              style={{
                transform: [
                  {
                    rotateZ: springValue.interpolate({
                      inputRange: [0, 1, 2, 3],
                      outputRange: ['45deg', '45deg', '45deg', '0deg'],
                    }),
                  },
                ],
              }}
            >
              <FAwesomeIcon name="plus" size={35} color="#FFF" />
            </Animated.View>
          </TouchableOpacity>
        </Animated.View>
      </View>
    );
  }
}

AddButtonForIos.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
};

export default withNavigation(AddButtonForIos);
