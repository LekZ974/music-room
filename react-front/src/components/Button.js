import React from 'react';

import { Button as NBButton } from 'native-base';

const Button = props => {
  return <NBButton {...props} />;
};
export default Button;
