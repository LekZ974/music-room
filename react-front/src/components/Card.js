import React from 'react';

import { Card as NBCard } from 'native-base';

const Card = props => {
  return <NBCard {...props} />;
};
export default Card;
