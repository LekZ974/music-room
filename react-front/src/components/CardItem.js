import React from 'react';

import { CardItem as NBCardItem } from 'native-base';

const CardItem = props => {
  return <NBCardItem {...props} />;
};
export default CardItem;
