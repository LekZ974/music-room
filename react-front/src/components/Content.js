import React from 'react';

import { Content as NBContent } from 'native-base';

const Content = props => {
  return <NBContent {...props} />;
};
export default Content;
