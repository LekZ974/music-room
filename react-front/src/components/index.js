export { default as ActivityIndicator } from './ActivityIndicator';
export { default as Button } from './Button';
export { default as Card } from './Card';
export { default as CardItem } from './CardItem';
export { default as Container } from './Container';
export { default as Content } from './Content';
export { default as Header } from './Header';
export { default as Text } from './Text';
export { default as TextInput } from './TextInput';
export { default as Title } from './Title';
export { default as View } from './View';
