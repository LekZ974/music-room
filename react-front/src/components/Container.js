import React from 'react';

import { SafeAreaView, StatusBar, Platform, View } from 'react-native';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import ActivityIndicator from './ActivityIndicator';

const centerView = {
  flexDirection: 'row',
  alignItems: 'center',
  justifyContent: 'center',
};

export const Container = ({ children, isLoading, grey, center, ...props }) => {
  return (
    <SafeAreaView
      style={{
        flex: 1,
        backgroundColor: grey ? '#FAFAFA' : 'white',
        marginTop: Platform.OS !== 'ios' ? StatusBar.currentHeight : 0,
      }}
      {...props}
    >
      <ActivityIndicator animating={isLoading} />
      <View
        style={{
          flex: 1,
          padding: 10,
          backgroundColor: grey ? '#FAFAFA' : 'white',
          ...(center && centerView),
        }}
      >
        {children}
      </View>
    </SafeAreaView>
  );
};

Container.defaultProps = {
  grey: false,
  center: false,
};

Container.propTypes = {
  children: PropTypes.node.isRequired,
  isLoading: PropTypes.bool.isRequired,
  grey: PropTypes.bool,
  center: PropTypes.bool,
};

export default connect(({ application }) => ({
  isLoading: application.isLoading,
}))(Container);
