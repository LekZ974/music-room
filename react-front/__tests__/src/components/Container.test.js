import React from 'react';

import renderer from 'react-test-renderer';
import { Container } from '../../../src/components/Container';

jest.mock('../../../src/components/ActivityIndicator', () => 'Activity indicator')

it('renders correctly', () => {
  const tree = renderer.create(<Container isLoading={false}>Snapshot test with CardItem Component!</Container>).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly with loading', () => {
  const tree = renderer.create(<Container isLoading>Snapshot test with CardItem Component!</Container>).toJSON();

  expect(tree).toMatchSnapshot();
});
