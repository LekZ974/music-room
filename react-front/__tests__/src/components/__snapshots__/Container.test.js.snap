// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "marginTop": 0,
    }
  }
>
  <Activity indicator
    animating={false}
  />
  <View
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
        "padding": 10,
      }
    }
  >
    Snapshot test with CardItem Component!
  </View>
</RCTSafeAreaView>
`;

exports[`renders correctly with loading 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "marginTop": 0,
    }
  }
>
  <Activity indicator
    animating={true}
  />
  <View
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
        "padding": 10,
      }
    }
  >
    Snapshot test with CardItem Component!
  </View>
</RCTSafeAreaView>
`;
